Backend/TemporalDataManager.py

```diff
@@ -0,0 +1,115 @@
+    #       fromdate: the datetime to search from
+    #       todate: the datetime to search to
+    #       days, hours, minutes, seconds = the time step of each unit
+    #       return value: return a csv format string in format:  datetime,val1,val2,val3
+    def get_trends_snapshot_as_csv(self, trends, fromdate, todate, woeid=0, days=0, hours=2, minutes=0, seconds=0):
+        # call algo manager to query to database and return trends within the time frame
+        # snapres return an array of database request for trend_snapshot data
+        # each element are for each trend in the trends argument
+        snapsres = self.algo.get_trends_snapshot_from_database(trends, fromdate, todate, woeid)
+
+        # preload the precv dictionary
+        precsv = {}             # dictionary to be parsed into csv. contain date and array of tweet volume values
+        maxvals = len(trends)   # max amount of value for each dictionary entry
+        tempdate = fromdate
+        while tempdate < todate:
+            precsv[tempdate] = []
+            tempdate += timedelta(days=days, hours=hours, minutes=minutes, seconds=seconds)
+        for csventry in precsv:
+            for i in range(maxvals):
+                precsv[csventry].append(0)
+
+        i = 0
+        for key in snapsres.keys():
+            rows = snapsres[key]
+            lenrows = len(rows)
+
+            print("Snapshots retrieved for {}: {}".format(trends[i], lenrows))
+
+            snapsbucket = self.algo.get_snapstime_bucket_from_database_tuples(rows, fromdate, days=days, hours=hours, minutes=minutes, seconds=seconds)
+            prevval = 0
+            for date in snapsbucket.keys():
+                snaps = snapsbucket[date]
+                totalvol = 0
+                for snap in snaps:
+                    totalvol += snap[4]
+                precsv[date][i] = totalvol    # updating the tweet volume value normally with just totalvol
+
+                # code comment below is for saving previous totalvol value.
+                # if new value is 0 then reuse the previous totalvol value
+                """
+                if totalvol == 0:
+                    precsv[date][i] = prevval
+                else:
+                    precsv[date][i] = totalvol
+                    prevval = totalvol
+                """
+            i += 1  # increment the trend index
+            print()
+
+        # create csv format string  datetime,val1,val2,val3....  to return to front end for grapahing
+        #       format: datetime,v1,v2,v3
+        #               datetime,v1,v2,v3
+        csv = ""
+
+        for key in precsv:
+            csv += "{},".format(key)
+
+            vals = precsv[key]
+            valcounter = 0
+            for val in vals:
+                valcounter += 1
+                csv += str(val)
+                if valcounter != maxvals:
+                    csv += ","
+
+            csv += "<br>"
```

Comment:
why this rather than \n? I thought csv text data normally didn't include markdown content.

Reply:
I wasnt sure if the data would come back with \n or not because doing \n doesnt visually do next line on the html return

Reply:
What do you mean by 'the html return'? The client just receives a text string, which can include the escape character \n. After all, that's what files have for new lines, and the csv format is mostly used for loading data from files.